.github/workflows/ca-clone-sequential-test.yml

```diff
@@ -984,6 +984,481 @@ jobs:
+          docker exec primary pki-server ca-range-generator-update --type cert legacy2
+          docker exec primary pki-server start --wait
+
+      - name: Check cert range objects
+        run: |
+          tests/ca/bin/ca-cert-range-objects.sh primaryds | tee output
+
+          # cert ranges should remain the same
+          cat > expected << EOF
+          SecurePort: 8443
+          beginRange: 31
+          endRange: 48
+          host: secondary.example.com
+
+          SecurePort: 8443
+          beginRange: 19
+          endRange: 48
+          host: primary.example.com
+
+          EOF
+
+          diff expected output
+
+      - name: Check cert next range
+        run: |
+          tests/ca/bin/ca-cert-next-range.sh primaryds | tee output
+
+          # cert nextRange should remain the same
+          cat > expected << EOF
+          nextRange: 49
+          EOF
+
+          diff expected output
+
+      - name: Switch secondary to legacy2
+        run: |
+          docker exec secondary pki-server stop
+          docker exec secondary pki-server ca-range-generator-update --type request legacy2
+          docker exec secondary pki-server ca-range-generator-update --type cert legacy2
+          docker exec secondary pki-server start --wait
+
+      - name: Check request range config in primary CA
+        run: |
+          tests/ca/bin/ca-request-range-config.sh primary | tee output
+
+          cat > expected << EOF
+          dbs.beginRequestNumber=11
+          dbs.endRequestNumber=15
+          dbs.requestCloneTransferNumber=5
+          dbs.requestIncrement=10
+          dbs.requestLowWaterMark=5
+          EOF
+
+          diff expected output
+
+      - name: Check request range config in secondary CA
+        run: |
+          tests/ca/bin/ca-request-range-config.sh secondary | tee output
+
+          cat > expected << EOF
+          dbs.beginRequestNumber=21
+          dbs.endRequestNumber=30
+          dbs.requestCloneTransferNumber=5
+          dbs.requestIncrement=10
+          dbs.requestLowWaterMark=5
+          EOF
+
+          diff expected output
+
+      - name: Check cert range config in primary CA
+        run: |
+          tests/ca/bin/ca-cert-range-config.sh primary | tee output
+
+          cat > expected << EOF
+          dbs.beginSerialNumber=0x13
+          dbs.endSerialNumber=0x30
+          dbs.serialCloneTransferNumber=0x9
+          dbs.serialIncrement=0x12
+          dbs.serialLowWaterMark=0x9
+          EOF
+
+          diff expected output
+
+      - name: Check cert range config in secondary CA
+        run: |
+          tests/ca/bin/ca-cert-range-config.sh secondary | tee output
+
+          cat > expected << EOF
+          dbs.beginSerialNumber=0x31
+          dbs.endSerialNumber=0x48
+          dbs.serialCloneTransferNumber=0x9
+          dbs.serialIncrement=0x12
+          dbs.serialLowWaterMark=0x9
+          EOF
+
+          diff expected output
```

Comment:
If I get the math correct the range in CS.cfg are smaller then in DS and not full used but the start range are always correct in case of gap.  The ID between 0x42 and 0x48 are lost because when a new range is computed it start from the value recorded in nextRange and do not take in account the range objects and/or configuration range so overlap should not be possible. Using the full recorded range we are requesting to utilise the full range.